public/app/features/dashboard/components/DashboardSettings/AutoRefreshIntervals.test.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import { defaultIntervals } from '@grafana/ui/src/components/RefreshPicker/RefreshPicker';
+
+import { AutoRefreshIntervals, getValidIntervals, Props, validateIntervals } from './AutoRefreshIntervals';
+import { TimeSrv } from '../../services/TimeSrv';
+
+const setupTestContext = (options: Partial<Props>) => {
+  const defaults: Props = {
+    renderCount: 0,
+    refreshIntervals: ['1s', '5s', '10s'],
+    onRefreshIntervalChange: jest.fn(),
+    getIntervalsFunc: intervals => intervals,
+    validateIntervalsFunc: () => null,
+  };
+
+  const props = { ...defaults, ...options };
+  const { rerender } = render(<AutoRefreshIntervals {...props} />);
+
+  return { rerender, props };
+};
+
+describe('AutoRefreshIntervals', () => {
+  describe('when component is mounted with refreshIntervals', () => {
+    it('then supplied intervals should be shown', () => {
+      setupTestContext({ getIntervalsFunc: () => ['5s', '10s'] }); // remove 1s entry to validate we're calling getIntervalsFunc
+
+      const input = (screen.getByRole('textbox') as unknown) as HTMLInputElement;
+
+      expect(input.value).toEqual('5s,10s');
+    });
+  });
+
+  describe('when component is mounted without refreshIntervals', () => {
+    it('then default intervals should be shown', () => {
+      setupTestContext({ refreshIntervals: (null as unknown) as string[] });
+
+      const input = (screen.getByRole('textbox') as unknown) as HTMLInputElement;
+
+      expect(input.value).toEqual('5s,10s,30s,1m,5m,15m,30m,1h,2h,1d');
+    });
+  });
+
+  describe('when component is updated from Angular', () => {
+    it('then intervals should be updated', () => {
+      const { rerender, props } = setupTestContext({});
+      const newProps = { ...props, renderCount: 1, refreshIntervals: ['2s', '6s', '11s'] };
+
+      rerender(<AutoRefreshIntervals {...newProps} />);
+
+      const input = (screen.getByRole('textbox') as unknown) as HTMLInputElement;
+
+      expect(input.value).toEqual('2s,6s,11s');
+    });
+  });
+
+  describe('when input loses focus and intervals are valid', () => {
+    it('then onRefreshIntervalChange should be called', () => {
+      const { props } = setupTestContext({ validateIntervalsFunc: () => null });
+
+      const input = (screen.getByRole('textbox') as unknown) as HTMLInputElement;
+
+      act(() => {
```

Comment:
`fireEvent` is internally wrapped in `act` so using it shouldn't be necessary here.

Reply:
Will fix!